### PR TITLE
Minimum setup project structure and build

### PR DESCRIPTION
**Pull request by TerraVenil**
As we agreed in Slack this PR contains minimum setup project structure and build.
Job that pushes a package to PyPi and a job that is responsible to docs generations will be added later.

### CODE REVIEW

**Review by Salgi15 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -6,6 +6,23 @@ with tket, a quantum computing toolkit and optimising compiler developed by Quan
 [QuEST](https://quest.qtechtheory.org/) is an open-source high performance simulator of
 quantum circuits, state-vectors and density matrices.
```
Comment (ferbetanzo) [inline README.md:8]: Could we use a structure for the README.md similar to this one: https://github.com/CQCL/pytket-qulacs/blob/develop/README.md ? We can do that in another PR though.

**Review by ferbetanzo — APPROVED** [review]
